src/txn/txn.c

```diff
@@ -605,10 +605,13 @@ __txn_commit_timestamp_validate(WT_SESSION_IMPL *session)
 		WT_RET_MSG(session, EINVAL, "no commit_timestamp required and "
 		    "timestamp set on this transaction");
 
+	commit_ts_keys = F_ISSET(txn,
+			     WT_TXN_TS_COMMIT_ALWAYS | WT_TXN_TS_COMMIT_KEYS);
+	durable_ts_keys = F_ISSET(txn, WT_TXN_TS_DURABLE_ALWAYS);
 	/*
 	 * If we're not doing any key consistency checking, we're done.
 	 */
-	if (!F_ISSET(txn, WT_TXN_TS_COMMIT_KEYS))
+	if (!commit_ts_keys && !durable_ts_keys)
```

Comment:
@sueloverso, I have changed this logic to perform key consistent check even if WT_TXN_TS_COMMIT_ALWAYS only is configured. I would like to know your opinion on this change. Do you think it is ok, or should we do key consistency check only if WT_TXN_TS_COMMIT_KEYS is configured?

Reply:
The reason that `key_consistent` is an explicit choice for commit is that it adds a lot of overhead. The `always` flag simply checks a bit. I think the `key_consistent` check is a level lower/deeper than just "this table should always/never use commit timestamps". I'd prefer to keep it separate. You may want to consider adding in a `key_consistent` setting for the durable timestamp too.